dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/MasterTaskExecutor.java

```diff
@@ -100,6 +101,15 @@ public void run() {
             TaskInstanceLogHeader.printInitializeTaskContextHeader();
             initializeTask();
 
+            if (DRY_RUN_FLAG_YES == taskExecutionContext.getDryRun()) {
+                taskExecutionContext.setCurrentExecutionStatus(TaskExecutionStatus.SUCCESS);
+                taskExecutionContext.setEndTime(System.currentTimeMillis());
+                MasterTaskExecutorHolder.removeMasterTaskExecutor(taskExecutionContext.getTaskInstanceId());
+                logicTaskInstanceExecutionEventSenderManager.successEventSender().sendMessage(taskExecutionContext);
+                log.info(
+                        "The current execute mode is dry run, will stop the logic task and set the taskInstance status to success");
+                return;
+            }
```

Comment:
It's better to move this after `beforeExecute` I am not sure if the statemachine will throw exception, since the task lifecycle might miss `running` event.

Reply:
> It's better to move this after `beforeExecute` I am not sure if the statemachine will throw exception, since the task lifecycle might miss `running` event, so the task state will be dispatched, and dispatched task cannot be converted to success.

I find the implementation in `WorkerTaskExecutor`,this is before `beforeExecute`,

Reply:
Test Result：
<img width="1192" alt="image" src="https://github.com/user-attachments/assets/b08c441f-6a26-4214-9d0d-84a11cf5012c">

Reply:
I have change, test is ok

<img width="1664" alt="image" src="https://github.com/user-attachments/assets/e2e22ad2-cda7-4e51-90f2-64adaa522f10">
<img width="1129" alt="image" src="https://github.com/user-attachments/assets/6d85f4bc-b522-4477-b1d8-7c7847678493">